index.js

```diff
@@ -127,6 +127,76 @@ function reset () {
   moduleAliases = {}
 }
 
+//
+// Functions to load configurations from a file
+//
+
+// Load configuration from any file
+function loadConfigFile (configPath) {
+  try {
+    var config = require(configPath)
+
+    // ES6 export default
+    if (config && config.default) {
+      return config.default
+    }
+
+    return config
+  } catch (e) {
+    // Do nothing
+  }
+}
+
+// Load configuration from the package.json file
+function loadPackageJSONFile (configPath) {
+  try {
+    var config = require(configPath)
+
+    if (config && config['module-alias']) {
+      return config['module-alias']
+    }
+
+    return {}
+  } catch (e) {
+    // Do nothing
+  }
+}
+
+function loadConfig (base, options) {
+  var configPath = ''
+  var config
+
+  // If a path has been provided, try loading the configuration using it
+  // It could be a simple JS, JSON or any file, or a package.json file
+  if (options.base) {
+    configPath = options.base
+
+    if (configPath.indexOf('package.json') > -1) {
```

Comment:
As of today, you can point to a directory, not necessarily a `package.json` directly ; see previous code: `base.replace(/\/package\.json$/, '') + '/package.json'` . Maybe keep something similar ? In that case you'd need to try loading the config from a package.json ; if it's empty, then try from the config file.

Reply:
The idea is to give more precedence to **module-alias.config.js** first before package.json.

custom file location > module-alias.config.js > package.json

You can read the discussion to find out more about it. But I'm open to make any modification. 

Thoughts?

Reply:
I agree with the order !

Here I mostly want to point out that when you pass a base path, there are three possibilities:
- It's a `/path/to/package.json` -> in that case, loadPackageJSONFile would do the job
- It's a `/path/to/a/config/file.json` -> in that case, loadConfigFile is the best
- It's a `/path/to/a/directory` -> Then I think we should do something with it. One way of approaching it is: try and load from a `module-alias.config.js` in that directory ; if it fails, try and load from a `package.json` in that directory. If it fails, maybe throw.

Reply:
@Kehrlann, agreed. I think I totally missed the third case. Will make it happen.

Reply:
Stale conversation, resolving